kcards/views/room.py

```diff
@@ -1,13 +1,19 @@
 from flask import Blueprint, Response, render_template
+from flask_api import status, exceptions
 
+from . import api_rooms
+from ..models import Room
 
-blueprint = Blueprint('room', __name__)
+blueprint = Blueprint('room', __name__, url_prefix='/rooms')
 
 
-@blueprint.route("/rooms/<code>")
+@blueprint.route("/<code>")
 def detail(code):
-    # Placeholder for how to 404 when a room is not in the room list.
-    # if code is None:
-    if code == 'not_found':
-        code = None
-    return Response(render_template("room.html", room=code))
+    room = Room.objects(code=code).first()
+
+    if not room:
+        raise exceptions.NotFound
+
+    response = Response(render_template("room.html", room=room))
+
+    return response, status.HTTP_200_OK
```

Comment:
You should configure your editor to always insert a newline at the end of files: http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline